src/Symfony/Component/Console/SignalRegistry/SignalMap.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\SignalRegistry;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class SignalMap
+{
+    private static array $map;
+
+    public static function getSignalName(int $signal): ?string
+    {
+        if (!\extension_loaded('pcntl')) {
+            return null;
+        }
+
+        if (!isset(self::$map)) {
+            $r = new \ReflectionExtension('pcntl');
+            $c = $r->getConstants();
+            $map = array_filter($c, fn ($k) => str_starts_with($k, 'SIG') && !str_starts_with($k, 'SIG_'), \ARRAY_FILTER_USE_KEY);
```

Comment:
what about harcoding the list instead? the list of signals and their name+value is known and standard, isn't it?

Reply:
not really :( Look at the comments: https://www.php.net/manual/en/pcntl.constants.php

Reply:
But then, don't we have an issue in tests where values are hardcoded instead of using the constants ?

Reply:
I used very stable ones